File: algos_assignment/2nd_grade/LeeJiYeon_dlwl1213/week1/10828.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int main() {
	stack <int> s;  //함수를 적용할 stack s를 생성해줌
	int N;  //명령의 수를 입력받기 위한 정수형 변수 N을 선언
	cin >> N;  //N을 입력받음

	for (int i = 0; i < N; i++) {  //입력받은 명령의 수만큼 반복하는 for문
		string str;  //명령은 string의 형식으로 입력받게 됨
		cin >> str;  //str에 명령을 입력받음
		if (str == "push") {  //명령이 push이면
			int a;  
			cin >> a;  //pusg해줄 정수 a를 입력받고
			s.push(a);  //stack에 push해줌
		}
		else if (str == "top") {  //명령이 top이면
			if (s.empty())   //우선 stack이 비어있는지 확인
				cout << "-1" << "\n";  //비어있다면 -1출력
			else  //stack에 원소가 있다면
				cout << s.top() << "\n";  //top을 출력
		}
		else if (str == "pop") {  //명령이 pop이라면
			if (s.empty())   //stack이 비어있는지 확인해서
				cout << "-1" << "\n";  //비어있으면 -1출력
			else {   //비어있지 않으면 top의 원소를 출력한 후 pop해줌
				cout << s.top() << '\n';
				s.pop();
			}
		}
		else if (str == "empty") {  //명령이 empty이면
			cout << s.empty() << "\n";  //empty함수를 통해 0,1의 값을 출력
		}
		else {  //마지막으로 명령이 size이면
			cout << s.size() << "\n";  //스택에 들어있는 정수의 개수를 출력
		}  
	}
	return 0;
}

File: algos_assignment/3rd_grade/seojinlee_metamong/2nd_week1/10828.py
import sys
def push(x):
    stack.append(x)
def pop():
    if(not stack):
        return -1
    else : 
        return stack.pop()

def size():
    return len(stack)

def empty():
    if len(stack)==0:
        return 1
    else:
        return 0

def top():
    if len(stack)==0:
        return -1
    else:
        return stack[-1]

n=int(sys.stdin.readline().rstrip())
stack=[]

for i in range(n):
    word=sys.stdin.readline().split()
    order=word[0]
    if order=="push": 
        push(word[1])
    elif order=="pop":
        print(pop())
    elif order=="size":
        print(size())
    elif order=="top":
        print(top())
    elif order=="empty":
        print(empty())
